protocols/ML01.md

```diff
@@ -0,0 +1,315 @@
+# ML01: Machine learning basic specification
+**THIS IS A DRAFT SPECFICATION AND IS NOT FINAL**
+
+| name                       | ML01               |
+|----------------------------|--------------------|
+| version                    | 0.1.0              |
+| status                     | proposal           |
+| author                     | Ryan Wilson (rw86) |
+| serving component(s)       | Machine Learning   |
+| consuming component(s)     | HCI                |
+| basic spec                 | yes                |
+| can be required by servers | no                 |
+| can be required by clients | HCI                |
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED",  "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119.
+
+
+## Changelog
+### Version 0.1
+
+- First draft
+
+# Foreword: Acknowledgements
+
+This specification is loosely based on ideas presented in an endpoint outline "ml_endpoints.md" written by Daphne Bogosian. There are some major architectural differences, outlined below:
+
+- The principal "backend should be responsible for all matters of persistance" agreed during protocol meetings has been incorporated
+  - In particular results are returned to the frontend as files via backend (this was also agreeed)
+- The principal "HCI should talk directly to backend" has been incorporated.
+  - In particular there is no BE <-> ML communication.
+- The API has been generalised to not specifically reference training/models and instead refence the more general "jobs":
+  - Gives more flexibility to the implementation for data processing tasks that do not fit this pattern.
+  - Removes burden on client.
+  - Jobs referencing models are still posibile (see implementation notes)
+- Response format made consistent with BE01.
+- Spec now only uses GET and POST, as these are the request types with most widely available browser support.
+  - This allows the front end to not require a server component (this was a principle agreed during meetings).
+
+# Foreword: Required Reading
+
+The reader is assumed to be fully familiar with the contents of BE01.
+
+Reference is also made to the bundle files and report format of UI01.
+
+# Foreword: URL
+
+Within this document the URL `http://ml.endpoint/` will be used to indicate the base URL of the backend server's API endpoint.
+
+# Foreword: Matching notation
+
+The matching notation used in this specification is identical to that used in BE01.
+
+# Overview
```

Comment:
There is no support in here for recommending ML algorithms, which was a requirement from the contact in medicine.

Reply:
I saw this as just being a special job type offered by ML - just one that does not have a specific algorithm, whereby ML produces a report indicating possible correlations / suggested specific algorithms. I think you envision more of a way of "filtering" the algorithm list? This would be as simple as adding optional parameter(s) to the endpoint that lists job types. If this is what you're referring to then this seems a perfectly straightforward and clean change to me.